### PR TITLE
Add Processes/Groups page

### PR DESCRIPTION
**Pull request by apradillap**
Connects to #937 

### What does this PR do?

Show /processes with http://madrid.gobierto.dev/sandbox/participation_proccesses.html

### Pending

When we merge with the PR of public contribution containers we have to take out the number of contributions and participants in groups teaser.

### CODE REVIEW

File: app/views/gobierto_participation/processes/index.html.erb [hunk 1]
```diff
@@ -4,20 +4,44 @@
...
 
-<ul>
-  <% @processes.each do |process| %>
-    <li><%= link_to process.title, gobierto_participation_process_path(process.slug) %></li>
-  <% end %>
-</ul>
+      <div class="pure-u-1 pure-u-md-2-3 groups">
+        <div class="pure-g three_by_three">
+          <% @processes.each do |process| %>
+            <%= render 'process_teaser', process: process, meta: false %>
```
Comment (amiedes) [inline app/views/gobierto_participation/processes/index.html.erb]: I think the `meta: true` parameter can be removed. Is something only used in the sandbox.

File: app/views/gobierto_participation/processes/index.html.erb [hunk 2]
```diff
@@ -4,20 +4,44 @@
...
 
-<ul>
-  <% @groups.each do |group| %>
-    <li><%= link_to group.title, gobierto_participation_process_path(group.slug) %></li>
-  <% end %>
-</ul>
+      <div class="pure-u-1 pure-u-md-2-3 groups">
+        <div class="pure-g three_by_three">
+          <% @groups.each do |group| %>
+            <%= render 'group_teaser', group: group, meta: true %>
```
Comment (amiedes) [inline app/views/gobierto_participation/processes/index.html.erb]: same here

File: app/views/gobierto_participation/processes/_group_teaser.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<div class="pure-u-1 pure-u-md-1-3">
+  <%= link_to gobierto_participation_process_path(group.slug), class:"content_block light" do %>
+    <h3><%= group.title %></h3>
+    <p><%= group.body %></p>
+    <% if meta %>
+      <div class="meta right">
```
Comment (amiedes) [inline app/views/gobierto_participation/processes/_group_teaser.html.erb]: ¿Can this be implemented? If not, we use a `<% pending do %> ... <% end %>` helper in the views to hide content temporarily.

File: app/views/gobierto_participation/processes/_process_teaser.html.erb [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  <h3><%= link_to process.title, gobierto_participation_process_path(process.slug) %></h3>
+  <div class="meta">
+    <svg xmlns="http://www.w3.org/2000/svg" width="40" height="12" viewBox="0 0 40 12">
+      <path fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M473.55201,778 C472.783025,776.234095 471.026818,775 468.983333,775 C466.231114,775 464,777.238576 464,780 C464,782.761424 466.231114,785 468.983333,785 C470.812002,785 472.410617,784.011729 473.277582,782.538462 L478.689085,782.538462 C479.556049,784.011729 481.154665,785 482.983333,785 C484.812002,785 486.410617,784.011729 487.277582,782.538462 L492.689085,782.538462 C493.556049,784.011729 495.154665,785 496.983333,785 C499.735552,785 501.966667,782.761424 501.966667,780 C501.966667,777.238576 499.735552,775 496.983333,775 C494.939849,775 493.183642,776.234095 492.414657,778 L487.55201,778 C486.783025,776.234095 485.026818,775 482.983333,775 C480.939849,775 479.183642,776.234095 478.414657,778 L473.55201,778 Z" opacity=".7" transform="translate(-463 -774)"/>
+    </svg>
+    <%= t('.current_stage_html', current_stage: process.current_stage) %>
```
Comment (amiedes) [inline app/views/gobierto_participation/processes/_process_teaser.html.erb:7]: ¿Can you modify the `current_stage` method so when the process is already closed it returns the latest stage?

File: app/views/gobierto_participation/processes/index.html.erb [hunk 5]
```diff
@@ -4,20 +4,44 @@
   </strong>
 <% end %>
 
-<h1>Gobierto Participation processes index</h1>
+<%= content_for :body_attributes do %>
+  class="theme-participation gobierto_participation"
+<% end %>
 
-Processes:
+<div class="gradient-bg">
```
Comment (amiedes) [inline app/views/gobierto_participation/processes/index.html.erb:11]: image

The second row is a little bit tabbed to the right.

**Review by amiedes — CHANGES_REQUESTED** [review]

**Comment by amiedes** [comment]
One little thing: we have the custom of deploying to staging 🚀  when a PR is ready to be reviewed.

**Comment by ferblape** [comment]
I have just released this branch to staging so we can test it. Please, don't forget to do this before creating the PR.

**Comment by ferblape** [comment]
What?  "Compra de plantas para la ofi" es just a group. Why is it in both processes and groups?

screen shot 2017-09-20 at 08 41 19

You can check this in participacion.gobify.net.

File: app/views/gobierto_participation/processes/_process_teaser.html.erb (hunk 4 above)
Comment (apradillap) [reply app/views/gobierto_participation/processes/_process_teaser.html.erb:7]: I corrected the current_stage method, had the wrong comparators and already shows the correct information.

File: app/views/gobierto_participation/processes/index.html.erb (hunk 5 above)
Comment (apradillap) [reply app/views/gobierto_participation/processes/index.html.erb:11]: Fixed

**Comment by apradillap** [comment]
@amiedes @ferblape thanks for your comments, I changed what you asked for. 👍

File: app/controllers/gobierto_participation/processes_controller.rb [hunk 6]
```diff
@@ -1,16 +1,17 @@
+# frozen_string_literal: true
+
 module GobiertoParticipation
   class ProcessesController < GobiertoParticipation::ApplicationController
-
     def index
-      @processes = current_site.processes.process
-      @groups    = current_site.processes.group_process
+      @processes = current_site.processes.open
```
Comment (ferblape) [inline app/controllers/gobierto_participation/processes_controller.rb]: @apradillap you need to filter and show only published processes, and not processes in draft

File: app/controllers/gobierto_participation/processes_controller.rb [hunk 7]
```diff
@@ -1,16 +1,17 @@
...
+      @groups = current_site.processes.group_process
```
Comment (ferblape) [inline app/controllers/gobierto_participation/processes_controller.rb]: Same with groups

File: app/controllers/gobierto_participation/processes_controller.rb [hunk 8]
```diff
@@ -1,16 +1,17 @@
...
     end
 
     def show
       @process = GobiertoParticipation::Process.find_by!(slug: params[:id])
```
Comment (ferblape) [inline app/controllers/gobierto_participation/processes_controller.rb]: 1. Filter by Site
2. Filter by published processes

**Review by ferblape — CHANGES_REQUESTED** [review]
Also:

1. Can you remove the method `find_person` from the ProcessesController
2. In general, please review all the queries, you are not scoping by published items (news, events...)

**Comment by apradillap** [comment]
I've worked out the details you told me about. This PR only reflects the details of the processes page. More details are in https://github.com/PopulateTools/gobierto/pull/970

**Review by ferblape — APPROVED** [review]
👍

**Comment by ferblape** [comment]
Feel free to merge it when you consider, @apradillap. I don't want to cause you any trouble with your branches.

**Pull request merged**